### PR TITLE
user story #2362046: upgrade plugin from jdk11 to jdk17

### PR DESCRIPTION
**Pull request by RalucaOT**
[https://center.almoctane.com/ui/?p=1001%2F1002#/entity-navigation?entityType=work_item&id=2362046](https://center.almoctane.com/ui/?p=1001%2F1002#/entity-navigation?entityType=work_item&id=2362046)

Upgrade plugin from jdk11 to jdk17. 
I updated the versions in the Gradle files and I resolved associated problems related to these changes.

### CODE REVIEW

File: src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/searchresult/CustomSearchTextField.java [hunk 1]
```diff
@@ -196,7 +198,7 @@ public void mousePressed(MouseEvent e) {
         if (ApplicationManager.getApplication() != null) { //tests
             final ActionManager actionManager = ActionManager.getInstance();
             if (actionManager != null) {
-                EmptyAction.registerWithShortcutSet(IdeActions.ACTION_CLEAR_TEXT, CommonShortcuts.ESCAPE, this);
+                ActionUtil.wrap(IdeActions.ACTION_CLEAR_TEXT).registerCustomShortcutSet(CommonShortcuts.ESCAPE, this);
```
[the remark this answers is not in the document]
Comment (RalucaOT) [reply src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/searchresult/CustomSearchTextField.java:201]: I've just tested the changes and the ESC still works. 
I verified that pressing ESC clears the text, as expected and everything appears to be functioning correctly.

File: src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/Constants.java [hunk 2]
```diff
@@ -68,15 +68,15 @@ public class Constants {
     public static final String TAB_MY_WORK_TITLE = "My work";
 
     public static String getOctaneRemoveIcon() {
-        if (UIUtil.isUnderDarcula()) {
+        if (!JBColor.isBright()) {
```
Comment (vladbenchea) [inline src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/Constants.java:71]: why these changes? is the UIUtil class deprecated in java17?

File: src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/components/ConnectionSettingsComponent.java [hunk 3]
```diff
@@ -40,7 +40,7 @@
 import com.intellij.util.net.HttpConfigurable;
 import com.intellij.util.ui.JBUI;
 import com.intellij.util.ui.UIUtil;
-import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.StringUtils;
```
Comment (vladbenchea) [inline src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/components/ConnectionSettingsComponent.java:43]: is lang deprecated as well in java17? And lang3 is being used now?

**Review by pikachugb — APPROVED** [review]

File: src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/Constants.java (hunk 2 above)
Comment (RalucaOT) [reply src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/Constants.java:71]: image
Yes, it is. And I switched to JBColor as Intellij has suggested

File: src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/components/ConnectionSettingsComponent.java (hunk 3 above)
Comment (RalucaOT) [reply src/main/java/com/hpe/adm/octane/ideplugins/intellij/ui/components/ConnectionSettingsComponent.java:43]: yes, it is
image

**Review by vladbenchea — APPROVED** [review]
